src/languageservice/parser/jsonParser07.ts

```diff
@@ -233,6 +233,16 @@ export function asSchema(schema: JSONSchemaRef): JSONSchema {
   if (isBoolean(schema)) {
     return schema ? {} : { not: {} };
   }
+  if (schema === undefined) {
+    return schema;
+  }
+  if (typeof schema !== 'object') {
+    // we need to report this case as JSONSchemaRef MUST be an Object or Boolean
+    console.warn(`Wrong schema: ${JSON.stringify(schema)}, it MUST be an Object or Boolean`);
```

Comment:
What do we expect to happen when we log this ? Do we expect users to refer to the logs?

Reply:
That is can happen when user use not entirely valid JSON Schema. 
I not sure how we should report that to user. As that check can be called multiple times, so I don't want to spam user with multiple error/warning notifications.